### PR TITLE
sk-inet: Add IP TOS socket option

### PR DESCRIPTION
**Pull request by h0lyalg0rithm**
DUMP and restore the TOS option

### CODE REVIEW

**Comment by rst0git** [comment]
It looks good to me.

@h0lyalg0rithm Could you add a commit message that explains *why* do we want to checkpoint/restore the TOS option? (see [CONTRIBUTING.md](https://github.com/checkpoint-restore/criu/blob/criu-dev/CONTRIBUTING.md#describe-your-changes))

**Review by rst0git — APPROVED** [review]
LGTM

**Comment by avagin** [comment]
pls add a zdtm test.

**Comment by h0lyalg0rithm** [comment]
@avagin The updated the existing test to also test out the ip tos option.

**Comment by Snorch** [comment]
> Author: Suraj Shirvankar <surajshirvankar@gmail.com>
> ...
> Signed-off-by: Suraj Shirvankar <surajshirvankar@gmail.com>
> Signed-off-by: h0lyalg0rithm <surajshirvankar@gmail.com>

I believe second signed-off is excess. Can you please remove it?

**Comment by Snorch** [comment]
> I updated the existing test to also test out the ip tos option.

Please split test changes into separate patch. You might want to read https://github.com/checkpoint-restore/criu/blob/criu-dev/CONTRIBUTING.md#separate-your-changes

**Comment by h0lyalg0rithm** [comment]
@Snorch I split the commits into two separate commits and signed the commits.Not sure why there were two signed offs

File: test/zdtm/static/sock_ip_opts00.c [hunk 1]
```diff
@@ -108,3 +111,4 @@ int main(int argc, char **argv)
 		close(sk_confs[i].sk);
 	return exit_code;
 }
+
```
Comment (Snorch) [inline test/zdtm/static/sock_ip_opts00.c:114]: Please remove excess newline at the end of file.

**Comment by Snorch** [comment]
You need to fix https://github.com/checkpoint-restore/criu/actions/runs/4679289483/jobs/8291560101?pr=2116 linter errors

**Review by Snorch — APPROVED** [review]

**Pull request merged**